File: 백준/Silver/10828. 스택/스택.py
import sys
r = sys.stdin.readline
sys.setrecursionlimit(10**6)

def solution():
    n = int(r().rstrip())
    stk = []
    for _ in range(n):
        command = r().split()
        if command[0] == 'push':
            stk.append(int(command[1]))
        elif command[0] == 'top' and stk:
            print(stk[-1])
        elif command[0] == 'size':
            print(len(stk))
        elif command[0] == 'empty':
            if stk: print(0)
            else: print(1)
        else:
            if stk:
                print(stk.pop())
            else:
                print(-1)
        
    
solution()

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {

		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(bf.readLine());
        StringBuilder sb = new StringBuilder();
		Stack<Integer> stack = new Stack<Integer>();

		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(bf.readLine());
			String m = st.nextToken();
            
			if (m.equals("push")) {
				int c = Integer.parseInt(st.nextToken());
				stack.push(c);
			} else if (m.equals("pop")) {
				if (stack.isEmpty()) {
					sb.append("-1\n");
				} else
					sb.append(stack.pop() + "\n");

			} else if (m.equals("size")) {
				sb.append(stack.size() + "\n");
			} else if (m.equals("empty")) {
				if (stack.isEmpty()) {
					sb.append("1\n");
				} else {
					sb.append("0\n");
				}
			} else if (m.equals("top")) {
				if (stack.isEmpty()) {
					sb.append("-1\n");
				} else
					sb.append(stack.peek() + "\n");
			}
		}
        System.out.println(sb);
	}

}